content/blog/base-vs-magrittr-pipe/index.Rmd

```diff
@@ -0,0 +1,68 @@
+---
+output: hugodown::hugo_document
+slug: base-vs-magrittr-pipe
+title: Differences between the base R and magrittr pipes
+date: 2023-03-08
+author: Hadley Wickham
+description: >
+    A discussion of the (relatively minor) differences between the native R pipe, 
+    `|>`, and the magrittr pipe, `%>%`.
+photo:
+  url: https://unsplash.com/photos/4CNNH2KEjhc
+  author: Sigmund
+# one of: "deep-dive", "learn", "package", "programming", "roundup", or "other"
+categories: [other] 
+tags: [magrittr]
+---
+
+<!--
+TODO:
+* [x] Look over / edit the post's title in the yaml
+* [x] Edit (or delete) the description; note this appears in the Twitter card
+* [x] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
+* [x] Find photo & update yaml metadata
+* [x] Create `thumbnail-sq.jpg`; height and width should be equal
+* [x] Create `thumbnail-wd.jpg`; width should be >5x height
+* [x] `hugodown::use_tidy_thumbnails()`
+* [ ] Add intro sentence, e.g. the standard tagline for the package
+* [ ] `usethis::use_tidy_thanks()`
+-->
+
+**Note:** The following has been adapted from a section of the forthcoming second edition of [R for Data Science](https://r4ds.hadley.nz/) that had to be removed due to length limitations.
+
+## Pipes
+
+With the 4.1.0 release of R, a native pipe operator, `|>`, was introduced.
```

Comment:
Is there a canonical 'we introduced `|>` in R' post that we can link to?

Reply:
Could we add in a sentence or two here about why a native pipe operator was introduced? I think having a little of that motivation will help people understand how they might be thinking about what to use when, and why differences are important.

Reply:
There are a bunch of posts on the introduction of the native pipe, but (and I've looked), I can't find one that's "official," per se. Jumping Rivers' post, [New features in R 4.1.0](https://www.jumpingrivers.com/blog/new-features-r410-pipe-anonymous-functions/), Michael Browman gets points for posting about it when it was still in R-devel ([The New Base Pipe](https://michaelbarrowman.co.uk/post/the-new-base-pipe/)), very handy comparison table in an SO thread: [[What are the differences between R's new native pipe `|>` and the magrittr pipe `%>%`?](https://stackoverflow.com/questions/67633022/what-are-the-differences-between-rs-new-native-pipe-and-the-magrittr-pipe)](https://stackoverflow.com/a/72086492/785190).

Because there wasn't a canonical post from R Core, I wasn't sure where to link to. It's in the [R devel news](https://cran.r-project.org/doc/manuals/r-devel/NEWS.html), but there's no way to link directly to the new features section of the 4.1.0 release…

Reply:
Yeah, I think quoting from the news is the best way to go.